## Workflows


### Overview
-------------
The Vertebrate Genomes Pipelines in Galaxy are intended to allow a user to generate high-quality near error-free assemblies of species from a user's own data or from the GenomeArk database. Depending on the available data, different versions of this workflow are available. 


| Link | Workflow |  Inputs | Outputs |
|---|--------|-------|--------|
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Kmer_profiling:VGP?tab=info)      |**MerylDB generation**:<br>Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury.<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads [`fastq`] | 1. Meryl Database of kmer counts <br> 2. GenomeScope summary, models and plots |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Kmer_profiling_Trio_data:VGP?tab=info)      |**MerylDB generation Trio**:<br>Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury using parental and offspring datasets<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads [`fastq`] <br> 2. Paternal short-read Illumina sequencing reads [`fastq`] <br> 3. Maternal short-read Illumina sequencing reads [`fastq`]| 1. Meryl Database of kmer counts <br> 2. GenomeScope summary, models and plots |
| [Dockstore](https://dockstore.org/my-workflows/github.com/Delphine-L/iwc/Long_read_assembly_with_Hifiasm)      |**Long Read Assembly with Hifiasm**:<br>Generate phased assembly based on PacBio Hifi Reads<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads [`fastq`]   <br> 2. K-mer database [`meryldb`]  <br>  3. Genome profile summary generated by Genomescope [`txt`]  <br>  4. Name of first assembly  <br>  5. Name of second assembly | 1. Primary assembly  <br>  2. Alternate assembly  <br>  3. QC: Busco report for both assemblies  <br>  4. Merqury report for both assemblies  <br>  5. Assembly statistics for both assemblies  <br>  6. Nx PLot for both assemblies  <br>  7. Size plot for both assemblies  |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Long_read_assembly_with_Hifiasm_and_HiC_data:VGP?tab=info)      |**Long Read Assembly with Hifiasm and HiC**:<br>Generate phased assembly based on PacBio Hifi Reads using HiC data for phasing<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads [`fastq`]  <br>  2. Concatenated HiC forward reads  [`fastq`]  <br>  3. Concatenated HiC reverse reads  [`fastq`]  <br>  4. K-mer database [`meryldb`]  <br>  5. Genome profile summary generated by Genomescope [`txt`]  <br>  6. Name of first assembly  <br>  7. Name of second assembly |  1. Haplotype 1 assembly   <br>  2. Haplotype 2 assembly  <br>  3. QC: Busco report for both assemblies  <br>  4. Merqury report for both assemblies   <br>  5. Assembly statistics for both assemblies  <br> 6. Nx PLot for both assemblies  <br>  7. Size plot for both assemblies  |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Trio_assembly_with_Hifiasm:VGP?tab=info)      |**Long Read Assembly with Hifiasm and Trio data**:<br>Generate phased assembly based on PacBio Hifi Reads using parental Illumina data for phasing<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads [`fastq`]  <br>  2. Concatenated Illumina reads : Paternal [`fastq`]  <br>  3. Concatenated Illumina reads : Maternal  [`fastq`]  <br>  4. K-mer database [`meryldb`]  <br>  5. Paternal hapmer database [`meryldb`]  <br>  6. Maternal hapmer database [`meryldb`]  <br>  7. Genome profile summary generated by Genomescope [`txt`]  <br>  8. Name of first haplotype  <br>  9. Name of second haplotype | 1. Haplotype 1 assembly  <br>  2. Haplotype 2 assembly  <br>  3. QC: Busco report for both assemblies  <br>  4. Merqury report for both assemblies  <br>  5. Assembly statistics for both assemblies  <br>  6. Nx Plot for both assemblies  <br>  7. Size plot for both assemblies  |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Purgedups:VGP?tab=info)      |**Purge Duplications**:<br>Purge Duplicated kmers from the assembly<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads - trimmed [`fastq`]  <br>  2. Primary Assembly (hap1) [`fasta`]  <br>  3. Alternate  Assembly(hap2) [`fasta`]  <br>  4. K-mer database [`meryldb`]  <br>  5. Genomescope model parameters [`txt`]  <br>  6. Estimated Genome Size [`txt`]  <br>  7. Name of first haplotype  <br>  8. Name of second haplotype| 1. Haplotype 1 purged purged assembly  <br>  2. Haplotype 2 assembly  <br>  3. QC: Busco report for both assemblies  <br>  4. Merqury report for both assemblies  <br>  5. Assembly statistics for both assemblies  <br>  6. Nx Plot for both assemblies  <br>  7. Size plot for both assemblies |                                  
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Purgedups_custom_cutoffs_for_primary_assembly:VGP?tab=info)      |**Purge Duplications**:<br>Rerun Purge Duplicated kmers with custom cutoffs for the primary assembly<br><span class="badge badge-success">ILL-PE</span> | 1. Hifi long reads - trimmed [`fastq`]  <br>  2. Primary Assembly (hap1) [`fasta`]  <br>  3. Alternate  Assembly(hap2) [`fasta`]  <br>  4. K-mer database [`meryldb`]  <br>  5. Genomescope model parameters [`txt`]  <br>  6. Auto-Alignement Primary assembly [`paf`] (From the **Purge Duplications** workflow)  <br> 7. Cutoffs for Primary Assembly [`txt`]  <br>  8. PBCSTATS base coverage for Primary assembly [`tab`](From the **Purge Duplications** workflow)  <br>  9. Estimated Genome Size [`txt`]  <br>  10. Name of first haplotype  <br>  11. Name of second haplotype| 1. Haplotype 1 purged purged assembly  <br>  2. Haplotype 2 assembly  <br>  3. QC: Busco report for both assemblies  <br> 4. Merqury report for both assemblies  <br>  5. Assembly statistics for both assemblies  <br> 6. Nx Plot for both assemblies  <br> 7. Size plot for both assemblies |  
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Purgedups_custom_cutoffs_for_alternate_assembly:VGP?tab=info)      |**Purge Duplications**:<br>Rerun Purge Duplicated kmers with custom cutoffs for the alternate assembly<br><span class="badge badge-success">ILL-PE</span> | 1. Hifiasm Alternate assembly + sequences purged from the primary assembly  [`fasta`] (From the **Purge Duplications** workflow)   <br> 2. pbcstat base wig file for alternate assembly [`wig`] (From the **Purge Duplications** workflow)  <br> 3.Cutoffs file for alternate assembly [`txt`]  <br> 4. Estimated Genome Size [`txt`]  <br> 5.  K-mer database [`meryldb`]  <br> 6. Hifiasm Purged Primary assembly (From the **Purge Duplications** workflow)  <br> 7. Name of first haplotype  <br> 8. Name of second haplotype| 1. Haplotype 1 purged purged assembly   <br> 2. Haplotype 2 assembly  <br> 3. QC: Busco report for both assemblies  <br> 4. Merqury report for both assemblies  <br> 5. Assembly statistics for both assemblies  <br> 6. Nx Plot for both assemblies  <br> 7. Size plot for both assemblies |  
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Bionano_Scaffolding:VGP?tab=info)      |**Bionano Scaffolding**:<br>Scaffolding using Bionano optical map data<br><span class="badge badge-success">ILL-PE</span> | 1. Scaffolded assembly [`fasta`]   <br> 2.Bionano data  [`cmap`]   <br> 3. Estimated genome size [`txt`]  <br> 4. Phased assembly generated by Hifiasm [`gfa1`] | 1. Scaffolds  <br>  2.  Assembly statistics  <br> 3. Nx PLot  <br> 4. Size plot  |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/HiC_Scaffolding:VGP?tab=info)      |**HiC Scaffolding with Yahs**:<br>Scaffolding using HiC Data with YAHS<br><span class="badge badge-success">ILL-PE</span> | 1. Scaffolded assembly [`fasta`]  <br>  2. Concatenated HiC forward reads  [`fastq`]   <br> 3. Concatenated HiC reverse reads  [`fastq`]  <br> 4. Restriction enzyme sequence [`txt`]  <br>  5. Estimated genome size [`txt`] | 1. Scaffolds   <br> 2.  Assembly statistics  <br> 3. Nx PLot  <br>  4. Size plot  <br>  5. QC: Busco report  <br> 6. Pretext Maps before and after scaffolding |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/HiC_Scaffolding_with_Salsa2:VGP?tab=info)      |**HiC Scaffolding with Salsa2**:<br>Scaffolding using HiC Data with Salsa2<br><span class="badge badge-success">ILL-PE</span> | 1. Scaffolded assembly [`fasta`]  <br>  2. Concatenated HiC forward reads  [`fastq`]  <br>  3. Concatenated HiC reverse reads  [`fastq`]  <br> 4. Restriction enzyme sequence [`txt`]  <br>  5. Estimated genome size [`txt`] | 1. Scaffolds  <br>  2.  Assembly statistics  <br> 3. Nx PLot  <br>  4. Size plot  <br>  5. QC: Busco report  <br> 6. Pretext Maps before and after scaffolding |
| [Dockstore](https://dockstore.org/workflows/github.com/Delphine-L/iwc/Assembly_decontamination:VGP?tab=info)      |**Decontamination**:<br>Decontaminate scaffolded assembly<br><span class="badge badge-success">ILL-PE</span> | 1. Scaffolded assembly [`fasta`]  | 1. Decontaminated assembly <br>  2.  Contaminant list |