---
autotoc: false
---

<slot name="/events/gcc2023/header" />

# Program

GCC2023 consists of the main meeting, interactive training, and a [collaboration
fest (CoFest)](/events/gcc2023/cofest/). The main meeting and the training are
interleaved on the same days with the CoFest following.

Full details, including the scientific program and keynotes, will be posted as
soon as the abstracts for talks, posters, and demos have been reviewed.
Meanwhile, please contribute to the program by [submitting your own
abstract](/events/gcc2023/abstracts/).

## Calendar

Each day of the meeting, there will be three sessions, morning talks, mid-day
training, and afternoon talks, with breaks and lunch in between. CoFest is
largely a free-form event that will be organized on the spot.

Training events will consist of five parallel tracks, each session being 2.5
hours long. More details for each of the trainings will be provided as we
approach the conference.

<table>
  <tbody>
    <tr class="lead text-center" style="background-color: #79C;">
      <th style="min-width: 12%; padding: 0.2em;">Mon</th>
      <th style="min-width: 12%; padding: 0.2em;">Tue</th>
      <th style="min-width: 12%; padding: 0.2em;">Wed</th>
      <th style="min-width: 12%; padding: 0.2em;">Thu</th>
      <th style="min-width: 12%; padding: 0.2em;">Fri</th>
      <th style="min-width: 12%; padding: 0.2em;">Sat</th>
      <th style="min-width: 12%; padding: 0.2em;">Sun</th>
    </tr>
    <tr class="text-center" style="background-color: #BCE">
      <th>Jul 10</th>
      <th>Jul 11</th>
      <th>Jul 12</th>
      <th>Jul 13</th>
      <th>Jul 14</th>
      <th>Jul 15</th>
      <th>Jul 16</th>
    </tr>
    <tr style="border: 1px solid lightgray">
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #BEF;" rowspan="3">
        <span class="lead">CoFest</span>
      </td>
      <td style="background-color: #BEF;" rowspan="3">
        <span class="lead">CoFest</span>
      </td>
      <td style="background-color: #BEF;" rowspan="3">
        <span class="lead">CoFest</span>
      </td>
    </tr>
    <tr style="border: 1px solid lightgray">
      <td style="background-color: #95E06C;">
        <span class="lead"><strong>Training</strong></span><br/>
        <ol style="padding-left: 5px;">
            <li style="nth-child(odd): background-color: red">Machine Learning with Galaxy</li>
            <li>Galaxy Workflow development</li>
            <li>Galaxy + R</li>
            <li>Intro to Genomics</li>
            <li>Single Cell Transcriptomics</li>
        </ol>
      </td>
      <td style="background-color: #95E06C;">
        <span class="lead"><strong>Training</strong></span><br/>
        <ol style="padding-left: 5px;">
            <li>Galaxy Tool Development</li>
            <li>Admin Roundtable</li>
            <li>Galaxy Tips and Tricks</li>
            <li>Genome Annotation</li>
            <li>Cancer Genomics</li>
        </ol>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span><br/>
      </td>
      <td style="background-color: #95E06C;">
        <span class="lead"><strong>Training</strong></span><br/>
        <ol style="padding-left: 5px;">
            <li>Contributing to the GTN</li>
            <li>Dev Small Groups</li>
            <li>Microbiology</li>
            <li>Genome Assembly with VGP</li>
            <li>Galaxy for Human Genetics</li>
        </ol>
      </td>
    </tr>
    <tr style="border: 1px solid lightgray">
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
      <td style="background-color: #C3F73A;">
        <span class="lead">Conference dinner</span>
      </td>
      <td style="background-color: #8CE;">
        <span class="lead">Talks (TBD)</span>
      </td>
    </tr>
  </tbody>
</table>

## Keynote speakers

GCC2023 is excited to present three keynote speakers that collectively will
showcase how Galaxy is used to address challenges in biodiversity, biosecurity
and structural biology.


### BioDiversity

<table style="width: 100%">
  <tbody>
    <tr class="lead text-left" style="background-color: white">
      <td style="border: 0; width: 20%;">
        <img src="/images/events/gcc2023/carolyn-hogg.png" style="max-height: 200px;" alt="Carolyn Hogg" />
      </td>
      <td style="border: 0;">
        <strong>Dr. Carolyn Hogg</strong><br/>
        Senior Research Manager<br/>
        Australasian Wildlife Genomics Group<br/>
        School of Life and Environmental Sciences<br/>
        University of Sydney
      </td>
    </tr>
  </tbody>
</table>

I have been working on the conservation of threatened species for over
twenty-five years both in Australia and overseas. I am currently the Senior
Research Manager for the Australasian Wildlife Genomics Group in the Faculty of
Science. I have been working with the Save the Tasmanian Devil Program for the
past ten years; in addition to other species such as orange-bellied parrots,
koalas, bilbies, and woylies. Working closely with both my academic and
conservation management partners my vision is to create a conservation legacy
for Australia by changing the way we integrate science, management and policy;
to proactively promote species’ resilience in the face of a changing world. We
are achieving this by developing better tools and technologies to integrate
molecular genetics into real-time conservation management decisions.


### Structural Biology

<table style="width: 100%">
  <tbody>
    <tr class="lead text-left" style="background-color: white">
      <td style="border: 0; width: 20%;">
        <img src="/images/events/gcc2023/kate-michie.png" style="max-height: 200px;" alt="Kate Michie" />
      </td>
      <td style="border: 0;">
        <strong>Dr. Kate Michie</strong><br/>
        Senior Lecturer<br/>
        Structural Biology Facility<br/>
        Mark Wainwright Analytical Centre<br/>
        University of New South Wales
      </td>
    </tr>
  </tbody>
</table>

Dr. Michie is a Senior Lecturer at the University of New South Wales (in BABS
and the Structural Biology Facility MWAC)with over two decades of expertise in
Protein Structural Biology. She has completed her doctoral degree from the
University of Sydney and received prestigious fellowships from L'Oréal-UNESCO
and Marie Curie to continue her research at the Laboratory of Molecular Biology
in Cambridge UK (2006-2010). Her research primarily focuses on molecular biology
of filament forming proteins, with a particular interest in structural biology.
Dr  Michie has published papers in top scientific journals, including Nature,
Molecular Cell and PNAS, and has made significant contributions to the field of
bacterial cell division, in the field of filament forming proteins and more
recently in light harvesting proteins from Algae.

Dr. Michie has also been involved in various community roles within the field of
structural biology. She established the Structural Biology Facility at UNSW in
2019, which provides state-of-the-art resources for researchers in the field.
She is the coordinator for the Greater Sydney Collaborative Access Program for
MX beamtime at the Australian Synchrotron, the President of the Sydney Protein
Group, and established the Computational Structural Biology Workgroup at UNSW,
bringing together researchers across campus who use computational structural
biology resources.

Aside from her research, Dr. Michie is also uses Alphafold2 extensively, which
she has helped set up in the cloud at UNSW within 2 weeks of the code being
publicly released. She has worked closely with a wide range of researchers to
teach them how to use this powerful tool to advance their research, and has
already had a number of successful collaborative grants awarded for projects
impacted by Alphafold2. Dr Michie's drive and leadership in this area have been
instrumental in providing UNSW researchers with access to the computing
resources they need to stay competitive on a world stage.

### BioDiversity

<table style="width: 100%">
  <tbody>
    <tr class="lead text-left" style="background-color: white">
      <td style="border: 0; width: 20%;">
        <img src="/images/events/gcc2023/roberto-barrero.png" style="max-height: 200px;" alt="Roberto Barrero" />
      </td>
      <td style="border: 0;">
        <strong>Associate Professor Roberto Barrero Gumiel</strong><br/>
        Senior Bioinformatics Solutions Architect<br/>
        Academic Division<br/>
        Research Portfolio<br/>
        eResearch
      </td>
    </tr>
  </tbody>
</table>

A/Professor Roberto Barrero has joined the Queensland University of Technology
bringing his Bioinformatics expertise to contribute tackling real world problems
in Agriculture, Plant Biosecurity and Human Health. He has led the
bioinformatics analysis of a range of projects with applications in both the
agricultural and biomedical domains. He has formal training in molecular biology
and developed bioinformatics skills over 15 years working as part of large
international collaboration consortia including Human Invitational Consortium,
Rice Annotation Project, Cattle tick Genome Sequencing Consortium and the
International Barley Genome Consortium. He has conducted analysis and
comparative genomics in crop plants, pathogenic bacteria, viruses, animal
genomics and human diseases.

He has devised innovative strategies in experimental design in a range of large
genomics initiatives in model and non-model species. He led the implementation
of customised computational workflows to data analysis to enable advanced
research applications, run bioinformatics workshops, and mentored and supervised
other bioinformaticians and PhD students.

More recently, he led a multidisciplinary team at the Plant Biosecurity
Cooperative Research Centre to develop a plant diagnostic toolkit that
accurately detects a range of viruses and viroids in a single test. Dr Barrero
and the team were awarded the ‘Biosecurity Impact Award 2018’ recognising their
measurable change to plant biosecurity by delivering research outcomes to
end-users and change in policy for the quarantine screening of viruses in
ornamental grasses. He has been invited to be the Project Coordinator of an
international collaboration involving National Plant Protection Organisations
from nine countries to harmonize standards and best practices for regulatory
application of high throughput technologies for phytosanitary diagnostics.
