{{> Events/GCCBOSC2018/Header }}

# Nominate a training topic!

**[Nominate a training topic Now](http://bit.ly/gccbosc2018-tr-nom)**

The joint [2018 Galaxy Community and Bioinformatics Open Source Conferences](https://gccbosc2018.sched.com/) start with training and *the training topics that will be offered **are determined by you***. Please take a few minutes to consider which training topics you would like to see offered at GCCBOSC2018. Any topics of interest to these communities can be nominated. 

[GCCBOSC2018](https://gccbosc2018.sched.com/) will be held 25-30 June in Portland, Oregon, United States. It will feature two days of training: the second of which is multi-track and will feature content for both the BOSC and Galaxy communities.  

Nominated topics can cover a wide range. For example:

- How to use open source software using just a web browser (e.g., Intro to Using Galaxy) 
- How to use software packages (The Newbie's Guide to BioRuby)
- Advanced applications of software (Genome Assembly with Galaxy, or Proteomics with BioPython)
- Software installation and configuration (Tool Wrapping for Galaxy, or Using CloudLaunch)
- Open source community (Building a Curation Community using Apollo, or Training using the Galaxy Training Network)

These are only examples.  If you think the communities would be interested in a topic, then please  [nominate it](http://bit.ly/gccbosc2018-tr-nom)!

Nominated topics will be published on this page as they come in.

**Topic Nominations close January 7.** Topics will be compiled into a uniform list by the GCCBOSC2018 organizers, and then voted on by the community starting one week later.

# Nominated Topics

Topics are posted here as they are nominated.  If you see a topic with a brief description that you would like to expand on then please send an email to outreach@galaxyproject.org.

As of December 15, 2017 these topics have been nominated:

## Practical use of the Galaxy API command line tools

How to use the Galaxy [API](https://galaxyproject.org/develop/api/) to automate workflows.

Galaxy has an always-growing API that allows for external programs to upload and download data, manage histories and datasets, run tools and workflows, and even perform admin tasks. This session will cover various approaches to access the API, in particular using the [BioBlend Python library](https://bioblend.readthedocs.io/en/latest/).

**Prerequisites**

* Unix command line
* Basic understanding of Galaxy from a developer point of view.
* Python programming.
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best.

## The Galaxy Database Schema

Running a production Galaxy server, you some times end up in a situation, where you manually need to interact with the database, e.g. you want to extract usage information, which can not be gathered from the built in reports tool. Or, a more  risky adventure: you need to change the state of a job to 'error'. For both cases, you require a good understanding of the Galaxy database schema. In this training session, you will learn some of the design concepts of the database, and how to extract (or if necessary change) information useful for a Galaxy admin.	Experience maintaining a production Galaxy server and a basic knowledge of relational databases and SQL statements.

**Prerequisites**

* Experience maintaining a production Galaxy server and a basic knowledge of relational databases and SQL statements

# Nominations in Progress

## GATK4

Training and presentations on the new version of GATK.

**Prerequisites**

* Sample input files

## Open workflow languages

Hands on with CWL/WDL and jobrunners.

**Prerequisites**

* Sample data

## Command line workflow management systems: Snakemake, Cluster Flow, Bpipe

How to use workflow management systems that are designed for the command line, such as [Snakemake](http://snakemake.readthedocs.io/en/stable/), [Cluster Flow](http://clusterflow.io/), and [Bpipe](https://github.com/ssadedin/bpipe#welcome-to-bpipe--).

**Prerequisites**

* Linux command line experience

## Analysis of scRNA sequencing data

How to analyze single cell RNA sequencing data, e.g. in [cellranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) or with dedicated R packages

**Prerequisites**

* TBD

