---
title: 2018 GCCBOSC2018 Conferences
---

{{> Events/GCCBOSC2018/Header }}


The joint [2018 Galaxy Community and Bioinformatics Open Source Conferences](https://gccbosc2018.sched.com/) start with training and *the training topics that will be offered **are determined by you***. 

[GCCBOSC2018](https://gccbosc2018.sched.com/) will be held 25-30 June in Portland, Oregon, United States. It will feature two days of training: the second of which is multi-track and will feature content for both the BOSC and Galaxy communities.  

# Topic Voting open through January 31

**[Vote Now!](https://docs.google.com/forms/d/e/1FAIpQLSdaRyU-5OuIbApSK20QRCB-Rlhhel70VHs3Cjvrm-Kyywi86A/viewform)**

Please take a few minutes to consider which training topics you would like to see offered at GCCBOSC2018.

# Nominated Topics

Topics were nominated by the community, and refined by potential instructors.  Topics are categorized by the **potential audience** and **the interface(s) to be used** in the workshop.

|  | Code | Explanation |
| --- | --- | --- |
| Audience | BB | Beginning Bioinformatics Analyst |
|                  | XB | Experienced Bioinformatics Analyst |
|                  | IP | Infrastructure Provider |
|                  | TD | Tool Developer |
| Interface | GG | Galaxy or other Graphical User Interface |
|                | CL | Command line, scripting and/or basic programming |

Q: What makes a bioinformatic analyst *experienced?* <br />
A: Attending at least one beginning bioinformatics session.

**The topics:**

| BB | XB | IP | TD |   | GG | CL |  | Topic                                    |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------------- |
| BB | XB | -  | -  | | -  | -  | | [Setting up for success: Everything you need to know when planning for an RNA-seq analysis](#setting-up-for-success-everything-you-need-to-know-when-planning-for-an-rna-seq-analysis) |
| BB | XB | -  | -  | | GG | -  | | [Galaxy 101 - A gentle introduction to Galaxy](#galaxy-101-a-gentle-introduction-to-galaxy) |
| BB | XB | -  | -  | | GG | -  | | [RNA-Seq Analysis in Galaxy](#rna-seq-analysis-in-galaxy) |
| BB | XB | -  | -  | | GG | -  | | [Galaxy for Power Users](#galaxy-for-power-users) |
| BB | XB | -  | -  | | GG | -  | | [How to analyze microbiota data with Galaxy](#how-to-analyze-microbiota-data-with-galaxy)  |
| BB | XB | -  | -  | | GG | -  | | [ChIPseq analysis using deepTools and MACS2](#chipseq-analysis-using-deeptools-and-macs2) |
| BB | XB | -  | -  | | GG | -  | | [Hi-C analysis in Galaxy](#hi-c-analysis-in-galaxy) |
| BB | XB | -  | -  | | GG | -  | | [Galaxy For Proteogenomics !](#galaxy-for-proteogenomics) |
| BB | XB | -  | -  | | GG | -  | | [Small genome de novo assembly using Galaxy](#small-genome-de-novo-assembly-using-galaxy) | 
| BB | XB | -  | -  | | GG | CL | | [GATK4: What's new and how to run it](#gatk4-whats-new-and-how-to-run-it) |
| BB | XB | -  | -  | | GG | CL | | [Data Carpentry Genomics Workshop: Data Organization and Automation with Shell](#data-carpentry-genomics-workshop-data-organization-and-automation-with-shell) |
| BB | XB | -  | -  | | GG | CL | | [Handling integrated biological data using Python (or R) and InterMine](#handling-integrated-biological-data-using-python-or-r-and-intermine) |
| -  | XB | IP | -  | | GG | -  | | [Bioinformatics Training and Education with the Galaxy Training Network](#bioinformatics-training-and-education-with-the-galaxy-training-network) |
| -  | XB | IP | -  | | -  | CL | | [Practical use of the Galaxy API command line tools](#practical-use-of-the-galaxy-api-command-line-tools) |
| -  | XB | IP | -  | | -  | CL | | [Building a Community Genome Database with Tripal v3](#building-a-community-genome-database-with-tripal-v3) |
| -  | XB | IP | TD | | -  | CL | | [Command line workflow management systems: Snakemake and Nextflow](#command-line-workflow-management-systems-snakemake-and-nextflow) |
| -  | XB | IP | TD | | -  | CL | | [WDL, the Workflow Description Language](#wdl-the-workflow-description-language) |
| -  | XB | IP | TD | | -  | CL | | [Community built analyses that run everywhere with bcbio](#community-built-analyses-that-run-everywhere-with-bcbio) |
| -  | XB | IP | TD | | -  | CL | | [Introduction to Common Workflow Language](#introduction-to-common-workflow-language) |
| -  | XB | IP | TD | | GG | CL | | [Deploying (Galaxy and your) applications into clouds](#deploying-galaxy-and-your-applications-into-clouds) |
| -  | -  | IP | TD | | -  | CL | | [Conda and Containers](#conda-and-containers) |
| -  | -  | IP | TD | | -  | CL | | [Writing & Publishing Galaxy Tools](#writing-and-publishing-galaxy-tools) |
| -  | -  | IP | -  | | -  | CL | | [Setting up a Galaxy instance as a service](#setting-up-a-galaxy-instance-as-a-service) |
| -  | -  | IP | -  | | -  | CL | | [Advanced customization of a Galaxy instance](#advanced-customization-of-a-galaxy-instance) |
| -  | -  | IP | -  | | -  | CL | | [Advanced accelerated Galaxy admin](#advanced-accelerated-galaxy-admin) |
| -  | -  | IP | -  | | -  | CL | | [Administration of Galaxy Infrastructures with Puppet](#administration-of-galaxy-infrastructures-with-puppet) |
| -  | -  | IP | -  | | -  | CL | | [Galaxy Interactive Environments](#galaxy-interactive-environments) |
| -  | -  | IP | -  | | -  | CL | | [Adding Galaxy Workflows to a Tripal Website](#adding-galaxy-workflows-to-a-tripal-website) |
| -  | -  | IP | -  | | -  | CL | | [The Galaxy Docker Project](#the-galaxy-docker-project) |
| -  | -  | IP | -  | | -  | CL | | [The Galaxy Database Schema](#the-galaxy-database-schema) |
| -  | -  | IP | -  | | -  | -  | | [Galaxy Architecture](#galaxy-architecture) |


<img class="pull-right" src="/src/events/gccbosc2018/training/rna-helix.jpg" width="60" alt="RNA-Seq" />

## Setting up for success: Everything you need to know when planning for an RNA-seq analysis

This workshop is geared towards researchers who are thinking about conducting an RNA-seq experiment and are interested in knowing more about what is involved. The planning process requires taking a step back to evaluate various factors and ultimately assess the feasibility of the experiment, expecting and avoiding potential pitfalls. 
â€‹
The workshop will go into detail about the different strategies for working with RNA-seq data depending on the biological question on hand:

* Best practice guidelines for experimental design (Biological replicates, Paired-end vs Single-end, Sequencing depth).
* Data storage and computational requirements.
* Overview of commonly used workflows for differential gene expression, de-novo assembly, isoform quantification and other uses of RNA sequencing.


The focus of this workshop is to outline current standards and required resources for the analysis of RNA sequencing data. This workshop will not provide an exhaustive list of software tools or pipelines available; rather it aims to provide a fruitful discussion on how best to prepare for performing RNA-seq data analysis from the lab to manuscript preparation.

**Prerequisites:**

* None


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Galaxy 101 - A gentle introduction to Galaxy

This workshop will focus on introducing the Galaxy user interface and how it can be used to analyze large datasets. We will cover the basic features of Galaxy, including where to find tools, how to import and use your data, and an introduction to workflows. This session is recommended for anyone who has not used, or only rarely uses Galaxy.

**Prerequisites:**

* Little or no experience using Galaxy.
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best. 


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## RNA-Seq Analysis in Galaxy

This workshop will introduce the concepts behind transcriptomics with NGS data and how to analyze this data in Galaxy. Specifically, this workshop will focus on de novo transcriptome reconstruction of RNA-seq data with the following goals:

* comprehensive identification of all transcripts across an experiment
* appropriately annotating classes of transcripts
* generating abundance estimates across a transcriptome
* significance testing of differentially expressed transcripts
* visualisation of reads and transcript structures

**Prerequisites:**

* a general knowledge of Galaxy (for example, you should be familiar with the material in Galaxy 101 or have attended Introduction to Galaxy).
* a wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best. 


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Galaxy for Power Users

Learn new tricks to optimize your research.

- scratch-book
- (propagating) tags
- history searching and filtering 
- collections and lists
- post-job triggers
- data libraries
- (hierarchical ) upload to collections
- other QoL tricks

**Prerequisites:**

* At least attendance of Galaxy 101.
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best. 


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## How to analyze microbiota data with Galaxy

The study of complex microorganism communities has been eased by the development of sequencing platforms and dedicated powerful bioinformatics tools. Several tools have recently been integrated into Galaxy for microbiota data analysis: Mothur, QIIME, MetaPhlAN, HUMAnN, FROGS, MEGAHIT, MetaSPAdes,...

In this training, we will show in this training how to analyze metagenomic and amplicon data inside Galaxy:

- Extraction of the OTUs using [QIIME](http://qiime.org/)/[Mothur](https://www.mothur.org/)
- Reconstruction of the taxonomic composition of a sample without OTUs using [MetaPhlAn](http://huttenhower.sph.harvard.edu/metaphlan2)
- Find the metabolic functions realized in an environment using [HUMAnN](https://huttenhower.sph.harvard.edu/humann)

**Prerequisites**

* Galaxy 101 or equivalent experience
* Ideally participants will already be familiar with the concepts behind metagenomics (e.g., OTU)
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## ChIPseq analysis using deepTools and MACS2

Did my IP work? Where is my signal? How well do my replicates correlate? What might my peaks even look like? Where are my peaks (or signal) in relationship to transcription start sites (or other features)? These are common questions that biologists first pose when dealing with ChIPseq data. We will use deepTools and MACS within Galaxy to demonstrate effective methods of

1. performing ChIPseq-specific quality control,
2. calling peaks and
3. visualising signal and peak enrichment around genes or other features.

We will also touch on aspects of ATAC-seq and techniques such as ribosomal profiling.

**Prerequisites**

- Galaxy 101 or equivalent experience.
- Ideally participants will already be familiar with generic NGS quality control and read mapping, since those won't be covered
- A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Hi-C analysis in Galaxy

This session will introduce the basics of chromosome conformation capture assays and their applications,
followed by best practices in mapping, QC'ing, visualazing and assigning 'topological associated domains' with Hi-C data.

**Prerequisites**

- Galaxy 101 or equivalent experience.
- Understanding of chromosome conformation capture (3C) and variants (HiC, 5C, 4C).
- Understanding of Illumina based "NGS".


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Galaxy For Proteogenomics !

Large-scale â€˜omicsâ€™ data generation is driven by high throughput genome and transcriptome sequencing, and proteome characterization using mass spectrometry. As a result, many researchers are turning to generating integrative analysis of these â€˜multi-omicsâ€™ datasets given the great potential to provide novel biological insights. These multi-omics applications are particularly challenging for data analysis, as they require the use of multiple, domain-specific software programs on scalable infrastructure capable of handling the computing and storage needs of this large-scale data.

Galaxy has been shown to solve the problems of software integration and scalability for multi-omics analysis, and additionally offer benefits for sharing complete workflows in a user-friendly environment accessible by wet-bench scientists. In this workshop, we will introduce the use of Galaxy platform for multi-omics data analysis applications. As a representative example, we will focus on the maturing field of proteogenomic applications. Proteogenomics most commonly integrates RNA-Seq data, for generating customized protein sequence databases, with mass spectrometry-based proteomics data, which are matched to these databases to identify novel protein sequence variants. (*Cancer Res.* (2017); 77(21):e43-e46. [doi: 10.1158/0008-5472.CAN-17-0331](http://cancerres.aacrjournals.org/content/77/21/e43.long).) 

*Workshop Content:* 

The course will include a basic introduction to proteomics and will include a hands-on session that will cover use of analytical workflows to generate proteomic databases from RNASeq data. The use and understanding of modules for analytical workflows for proteogenomics analysis will be discussed.

*Workshop Schedule:*

* Introduction to multi-omic studies
* RNASeq Data Processing: Generating protein sequence search databases using Galaxy platform
* Hands-on session for proteomics data analysis using Galaxy
* Identification of novel proteoforms and visualization
    
*Workshop Goals:*
    
* Introduce the Galaxy framework as a solution for data analysis across â€˜omicsâ€™ domains
* Demonstrate use of Galaxy for a proteogenomic analysis (RNA-seq and proteomic integrative analysis)
* Lay the foundation for attendees to implement Galaxy at their own facility or institution to meet â€˜omicsâ€™ data analysis and informatics needs

**Prerequisites:**

* Users will need laptop for hands-on training


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Small genome de novo assembly using Galaxy

Workshop will cover the basics of de novo genome assembly using a small genome example. This includes project planning steps, selecting fragment sizes, initial assembly of reads into fully covered contigs, and then assembling those contigs into larger scaffolds that may include gaps. The end result will be a set of contigs and scaffolds with sufficient average length to perform further analysis on, including genome annotation. This workshop will use tools and methods targeted at small genomes. The basics of assembly and scaffolding presented here will be useful for building larger genomes, but the specific tools and much of the project planning will be different.

**Prerequisites**

* Galaxy 101
* A wifi enabled laptop with a web browser


<img class="pull-right" src="/src/images/logos/gatk-logo-wrench.png" width="60" alt="GATK" />

## GATK4: What's new and how to run it

GATK4 is the brand new version of the Genome Analysis Toolkit (GATK), an open-source genomics software package focused on variant discovery. This workshop will highlight the key changes and updates in the new 4.0 version which was released in January 2018 (see https://software.broadinstitute.org/gatk/gatk4). 

The goal of the workshop is to equip participants with the essential know-how to get started with GATK4, whether they have previously used GATK or not. We will walk participants through hands-on exercises aimed at developing familiarity with the GATK4 command line. We will also guide participants through several different ways of running GATK4 tools and pipelines on publicly available platforms including but not limited to Galaxy. We hope this will help participants understand what are their options and choose the platform that best suits their needs and abilities.
 
The workshop will cover the following topics:

1. Introduction to GATK and the Best Practices: 
  - Purpose, variant calling basics and standard data types

2. What's new in GATK4:
  - New syntax/invocations, performance improvements and tips & tricks for using GATK effectively
  - Expanded scope of analysis: 
	- Scaling germline variant discovery with GenomicsDB
	- Calling somatic short variants with the new and improved Mutect2
	- Calling somatic copy number variants with GATK CNV

3. Options for running GATK 
  - Running tools individually on a laptop (with Docker)
  - Running Spark-capable tools individually on a Spark cluster (via Google Dataproc)
  - Running pipelines on a laptop using Cromwell
  - Running pipelines on Google Cloud using Cromwell + Pipelines API
  - Running pipelines on the FireCloud analysis portal through the web GUI 
  - Running pipelines on FireCloud through the API + Python bindings 
  - Running tools and pipelines on Galaxy

Free credits will be provided for running on Google Cloud. For more information on FireCloud, see https://software.broadinstitute.org/firecloud.

**Prerequisites:**

- Basic familiarity with terms and concepts of genetics and genomics, including high-level understanding of high-throughput sequencing technologies and file formats used in genomic analysis. 
- Basic familiarity with the command line environment and usage of command line tools.
- NO familiarity is expected with Spark or cloud computing concepts.

<img class="pull-right" src="/src/images/logos/DataCarpentryLogo.png" width="140" alt="Data Carpentry" />

## Data Carpentry Genomics Workshop: Data Organization and Automation with Shell

The goal of this **one-day** tutorial is to teach participants the fundamental data management and analysis skills and needed to conduct genomics research including: best practices for organization of bioinformatics projects and data as well as use of command line utilities for managing large volumes of genomic data. This tutorial is derived from the Data Carpentry Genomics Workshop, focused on the organization and intro to command line lessons. The tutorial uses active learning and hands-on practice to teach participants the skills and perspectives needed to work effectively with genomic data. Data Carpentryâ€™s aim is to teach researchers basic concepts, skills, and tools for working with data so that they can get more done in less time, and with less pain.

Lessons:

* http://www.datacarpentry.org/organization-genomics/
* http://www.datacarpentry.org/shell-genomics/

The tutorial assumes no prior experience with the tools covered. However, learners are expected to have some familiarity with biological concepts, including nucleotide abbreviations. Participants should bring their laptops and plan to participate actively.


<img class="pull-right" src="/src/images/logos/intermine-squareish-trans-300.png" width="55" alt="InterMine" />

## Handling integrated biological data using Python (or R) and InterMine

This tutorial will guide you through loading and analyzing integrated biological data (generally genomic or proteomic) in [InterMine](http://intermine.org/) via an API in Python or R. Topics covered will include automatically generating code to perform queries, customising the code to meet your needs, and automated analysis of sets, e.g gene sets, including enrichment statistics. Skills gained can be re-used in any of the dozens of InterMines available, covering a broad range of organisms and dedicated purposes, from model organisms to plants, drug targets, and mitochondrial DNA.

**Prerequisites**

* Basic Python or R skills are advantageous but not required. 
* WiFi-enabled laptop with Python or R Studio.


<img class="pull-right" src="/src/images/galaxy-logos/GTNStarBars300.png" width="60" alt="Galaxy Training Network" />

## Bioinformatics Training and Education with the Galaxy Training Network

Galaxy with its flexibility, reproducibility, and scalability is an ideal environment for teaching
and training diverse scientific topics. 

The Galaxy Training Network is a community initiative dedicated to high-quality Galaxy-based
training around the world. One of its objectives is to support trainers with complete training
material and recommendations about bioinformatics training. Templates and best training practices
were defined to help trainers create new material, unify the different material, and make training
materials more accessible and easy for users to learn and for teachers to teach (https://training.galaxyproject.org).

This workshop will introduce participants to the infrastructure of the GTN training materials
and describe how to generate training materials following best practices.
Participants will be introduced to the generation of Galaxy Interactive Tours
and Docker Galaxy flavors dedicated to teaching and training sessions.
The workshop will also cover best practices for running Galaxy-based workshops - how to plan a training session based on number of attendees, time constraints, resource availability, etc.

**Prerequisites**

* Basic familiarity with using Galaxy (how to import datasets and run tools).
* Basic familiarity with git and Docker will also be helpful for parts of the workshop.
* A WiFi-enabled laptop with a modern web browser. Google Chrome or Firefox will work best.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Practical use of the Galaxy API command line tools

How to use the Galaxy [API](https://galaxyproject.org/develop/api/) to automate workflows.

Galaxy has an always-growing API that allows for external programs to upload and download data, manage histories and datasets, run tools and workflows, and even perform admin tasks. This session will cover various approaches to access the API, in particular using the [BioBlend Python library](https://bioblend.readthedocs.io/en/latest/).

**Prerequisites**

* Unix command line
* Basic understanding of Galaxy from a developer point of view.
* Python programming.
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best.


<img class="pull-right" src="/src/images/logos/tripal-logo-dark.png" width="110" alt="Tripal" />

## Building a Community Genome Database with Tripal v3

[Tripal](https://tripal.info/) v3 is the newest version of a popular and open source genomics website construction software. Using Docker containers, we will install a Tripal v3 site (including Apache, PostGreSQL, and Drupal), to create an empty genome database and site. We will review how to load many data types used by many of the core and extension Drupal modules, including organisms, analyses, genes and genomes, functional annotations and controlled vocabularies. We will provide credits on the XSEDE cloud system Jetstream for this training, so students can follow along on their own laptops.

**Prerequisites:**

* Linux command line.

<div class="pull-right"><img src="/src/images/logos/snakemake-logo-trans.png" width="60" alt="Snakemake" /> &nbsp; <img src="/src/images/logos/nextflow-logo-no-text-trans.png" width="50" alt="Nextflow" />
</div>

## Command line workflow management systems: Snakemake and Nextflow

This session introduces command line and text based workflow management systems by the examples of Snakemake and Nextflow.  We will start with an overview of concepts and challenges then spend an hour on each of these platforms.

*Snakemake*

We will show how to define a workflow in the Snakemake workflow language, and how to execute it using the Snakemake command line interface. In particular, we will show how Snakemake enables reproducible science by allowing

* automation of every step of a data analysis from raw data to final figures
* scalability of the workflow to any major computing architecture (compute server, cluster, grid, cloud) without having to modify the workflow definition
* portability of the workflow by integration with the Conda package manager and Singularity containers.

*Nextflow*

This session will introduce the Nextflow framework, the tool basic concepts and how it enables the definition and the deployment of large-scale distributed computational pipeline in a portable and reproducible manner across cloud and clusters. In particular it will be discussed: 

* installation and introduction to the dataflow processing model
* workflows parallelisation and scalability
* portable workflows containerisation with Docker, Singularity and Shifter 
* cloud deployment strategies

**Prerequisites**

* Linux command line experience


<img class="pull-right" src="/src/images/logos/wdl-logo_white.png" width="95" alt="WDL" />

## WDL, the Workflow Description Language

The advent of open, portable workflow languages is an exciting development which allows for the definition of a workflow to be decoupled from the execution. One can create workflows which can run unmodified on local compute, HPC clusters, or the cloud. As these languages are not tied to a specific execution environment, the descriptions can easily be shared, discovered and even composed together to form more complex workflows. The [Workflow Description Language](http://openwdl.org/) (WDL) is an open, community driven standard that is designed from the ground up as a human-readable and -writable way to express portable tasks and workflows.

In this session weâ€™ll walk through the lifecycle of writing, sharing and discovering portable workflows in WDL. Weâ€™ll introduce the WDL syntax. Youâ€™ll learn how to write and run a workflow locally. Weâ€™ll demonstrate how to use [EPAMâ€™s Pipeline Builder tool](https://github.com/epam/pipeline-builder) to visualize WDL workflows. Weâ€™ll look at [Dockstore](https://dockstore.org/), an open platform where one can publish, share and discover workflows. Finally weâ€™ll put everything together and see how we can compose workflows we find on Dockstore together with our own additions to create new, more powerful workflows.

**Prerequisites:**

* Mac or Linux computer
* Java 8 installed and runnable from the command line
* A workshop bundle downloaded from TBD


<img class="pull-right" src="/src/images/logos/bcbio-logo-no-text.png" width="80" alt="bcbio" />

##  Community built analyses that run everywhere with bcbio

**[bcbio](http://bcb.io/)** is community built analyses for germline and somatic
variant calling, RNA-seq, single cell, smallRNA and ChIP-seq. This workshop
will focus on using bcbio to run analysis pipelines in heterogenous
environments -- local machines, HPC, cloud providers and commercial services
(and also hopefully Galaxy). Attendees will learn how to practically run their
analyses on their platform of choice, while discussing how the community can
contribute to building, sharing and maintaining workflows across multiple
platforms. Recent presentations of bcbio show [some of the topics we plan to cover](http://bcbio-nextgen.readthedocs.io/en/latest/contents/presentations.html#presentations).


<img class="pull-right" src="/src/images/logos/cwl-logo-small-trans.png" width="130" alt="CWL" />

## Introduction to Common Workflow Language

Common Workflow Language  (http://commonwl.org) is a standard for writing portable scientific workflows that can execute on a variety of compute environments and workflow systems.

* What is CWL, history
* Status of CWL implementations (Arvados, Toil, Rabix, Galaxy, Cromwell, cwtool, ...)
* Wrapping bioinformatics tools in CWL
* Connecting tools together into workflows
* Best practices for writing portable workflows
* Hands on "bring your own pipeline" session
* Q&A on advanced topics based on audience

**Prerequisites**

* Unix command line experience
* Experience with Docker


<img class="pull-right" src="/src/images/galaxy-logos/cloudlaunch-small-blue.png" width="120" alt="CloudLaunch" />

## Deploying (Galaxy and your) applications into clouds

This tutorial will have two parts. Part 1 will demonstrate how to use the all-new CloudLaunch service to launch and manage instances of Galaxy on the Cloud on multiple cloud providers. In part 2, we will cover the technical process of adding custom applications into CloudLaunch and making them available for launching on any supported cloud (AWS, Azure, GCE, OpenStack). This part will also cover use of the CloudLaunch API, enabling external applications to leverage CloudLaunch capabilities.

**Prerequisites**

* Part 1: a laptop with WiFi and a modern browser
* Part 2: Basic programming skills (Python, Angular 2/4/5 useful but not essential?)


<img class="pull-right" src="/src/images/logos/Conda_480.png" width="95" alt="Conda" />

## Conda and Containers

This workshop is aimed at people with some desire to develop dependencies for tools (either Galaxy or Common Workflow Language (CWL) tools).

We believe the best practice for declaring dependencies for either Galaxy or the CWL is using Conda and Bioconda. Conda is a cross platform package manager that has minimal requirements to use which makes it ideal for HPC. It can also build isolated environments ideal for platforms like Galaxy or CWL implementations. The Bioconda project is a set of Conda recipes for bioinformatics. The BioContainers project builds best practice containers automatically for all Bioconda recipes, so building a BioConda recipe for a package allows the same binaries to be used by both Galaxy (inside or outside a container) and by any conformant CWL implementation.

We will go through the process of creating, testing, and publishing a Bioconda package and we will work through an example of connecting these packages to a real world tool. Participants will be able to work through the examples using either Galaxy or CWL tools.

**Prerequisites**

* Some knowledge of tool development - either CWL or Galaxy.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Writing & Publishing Galaxy Tools

This session will walk developers and bioinformaticians through the process of taking a working script or application and turning it into a Galaxy tool. It will also cover the basics of using Planemo: a command-line utility to assist in building and publishing Galaxy tools. We will investigate wrapping, common parameters, tool linting, best practices, loading tools into Galaxy, citations, and publishing tools to Github and the Galaxy Tool Shed. Common tips and tricks will be discussed as well as insights from experienced tool developers.

**Prerequisites:**

* A general knowledge of Galaxy (for example, you should be familiar with the material in Galaxy 101 or have attended Introduction to Galaxy).
* Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what commands like cd, mv, rm, mkdir, chmod, grep can do then you will struggle in this workshop.
* A wi-fi enabled laptop with a modern web browser. Chrome or Firefox will work best.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Setting up a Galaxy instance as a service

In this workshop, you will learn what is important when you set up a Galaxy server from scratch, what are the pitfalls you might run into, how to interact with the potential users of the service you gonna offer, and how to make sure, the Galaxy instance you have set up is really used in the end. After a general introduction, several Galaxy installations are presented. The session will include some demonstrations and hands-on exercises. We will finish with a panel discussion, where we intend to discuss questions from the workshop participants.

**Prerequisites:**

* Familiar with the Bioinformatics problems (and their solutions) that wet lab scientists run into.
* Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Advanced customization of a Galaxy instance

Do you have your lab's Galaxy instance set up and configured but want to give it some more love without diving too deep into the code? This training will show you step by step how to modify some advanced but not complex parts of the installation. We will teach you how to:

* modify the menu
* prepare a custom tour
* adjust the graphical interface
* translate the UI labels to different language
* set up a built-in user/group chat
* write and activate interface webhooks 

**Prerequisites**

* Introduction to Galaxy admin: Setting up a Galaxy instance as a service, or equivalent experience
* Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* A wi-fi enabled laptop with a modern web browser. Google Chrome, Firefox and Safari will work best.


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Advanced accelerated Galaxy admin

A compressed top level review of the advanced parts of the weeklong Galaxy Administrators Course delivered multiple times in the past 2 years. Given the size of the scope of this topic we will be explaining advanced concepts, pointing out resources and providing guidance, tips, and tricks rather than going through the exercises and into details.

* Introduction to Galaxy admin: Setting up a Galaxy instance as a service, or equivalent experience
* Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.


<img class="pull-right" src="/src/images/logos/puppet_logo_trans.png" width="100" alt="Puppet" />

## Administration of Galaxy Infrastructures with Puppet

Administering Galaxy infrastructures can be a daunting task. Configuration management allows one to implement infrastructure using code as opposed to checklists and HowTo guides in order to have reliable installations and the ability to replicate an installation. 

Building off of the work started at the GCC2017 Hackathon, this training will introduce the idea of using puppet for configuration management of systems using existing puppet code and other open source tools available that augment puppet. This session requires no prior experience with puppet or other configuration management tools. 

Draft training materials are [available online](https://github.com/millerjl1701/jedha-temple).

**Prerequisites:**

* You have setup a Galaxy instance beforeâ€¦  maybe.
* Knowledge and comfort with the Unix/Linux command line interface and use of a text editor.
* Vagrant with puppetlabs/centos-7.2-64-nocm box available or a CentOS 7 virtual machine (Infrastructure Server group install)
* **No** Puppet, Chef, or Ansible experience is required.

<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Galaxy Interactive Environments

In this session you will get in-depth introduction to Interactive Environments (IE). You will learn how to setup and secure IEâ€™s ([Jupyter](http://jupyter.org/), [RStudio](https://www.rstudio.com/), etc.) in a production Galaxy instance. Moreover, we will create an IE on-the-fly to get you started in creating your own Interactive Environments.In this session you will get an introduction to Interactive Environments (IE) as an easy and powerful way to integrate arbitrary interactive web services into Galaxy. We will demonstrate the [IPython Galaxy Project](https://github.com/bgruening/galaxy-ipython) and the general concept of IEâ€™s.

**Prerequisites:**

* Basic understanding of Galaxy from a developer point of view.
* General knowledge about Docker
* Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 


<img class="pull-right" src="/src/images/logos/tripal-logo-dark.png" width="110" alt="Tripal" />

## Adding Galaxy Workflows to a Tripal Website

Learn to integrate two popular and widely adopted open source [GMOD](https://gmod.org/) tools: [Tripal](https://tripal.info/), a content management system for building community genome websites, and Galaxy, a web-based workflow engine for biological data analysis. The new Tripal-to-Galaxy bridge allows users of community databases to select pre-designed workflows for common analyses and upload their own data or utilize site data as input. We will also use new R Markdown wrapped Galaxy tools (Aurora Galaxy Tools) to construct HTML reports summarizing and visualizing the workflow output.

**Prerequisites**

* Familiarity with Galaxy user interface and basic Galaxy server administration. 


<img class="pull-right" src="/src/images/logos/DockerInGalaxy.png" width="100" alt="Galaxy Docker Project" />

## The Galaxy Docker Project

In this session you will learn the internals of the Docker Galaxy Image.
We will show you tips and tricks on how to run the Galaxy Docker Image successfully in production,
how to manage updates and how to bind the container to a cluster scheduler.
Moreover, you will learn how to create your own Galaxy flavour mixing a variety of different tools and visualisations.
A composed Galaxy setup will be demonstrated and you will learn how you can orchestrate Galaxy to dynamically scale
with rising user demand.

**Prerequisites**

* Basic understanding of Galaxy from an admin point of view
* General knowledge about Docker
* Familiarity with Unix command line


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## The Galaxy Database Schema

Running a production Galaxy server, you some times end up in a situation, where you manually need to interact with the database, e.g. you want to extract usage information, which can not be gathered from the built in reports tool. Or, a more  risky adventure: you need to change the state of a job to 'error'. For both cases, you require a good understanding of the Galaxy database schema. In this training session, you will learn some of the design concepts of the database, and how to extract (or if necessary change) information useful for a Galaxy admin.

**Prerequisites**

* Experience maintaining a production Galaxy server
* Basic knowledge of relational databases and SQL statements


<img class="pull-right" src="/src/images/galaxy-logos/galaxy_project_logo_square_no_text.png" width="60" alt="Galaxy" />

## Galaxy Architecture

Want to know the big picture about what is going on inside Galaxy? This workshop will give participants a practical introduction to the Galaxy code base with a focus on changing those parts of Galaxy most often modified by local deployers and new contributors.

The workshop will include the following specific content:

* A description of the various file and top-level directories in the Galaxy code base.
* An overview of important Python modules - including models, tools, jobs, workflows, visualisations, and API controllers.
* An overview of important Python objects and concepts in the Galaxy codebase - including the Galaxy transaction object ("trans"), the application object ("app") , and the configuration object ("config").
* An overview of various plugin extension points. - An overview of important JavaScript modules that power the front-end.
* An overview of important JavaScript concepts used by Galaxy - in particular Backbone MVC, Webpack, ES6, and Vue.
* An overview of the client build system used to generate compressed JavaScript, cascading stylesheets, and other static web assets. 
* A demonstration of a complete start-to-finish modification of Galaxy - including forking the project on Github, modifying files, running the tests, checking style guidelines, committing the change, pushing it back to your local Github fork, and opening a pull request. 
* A brief description of other projects in the Galaxy ecosystem (CloudMan, the Tool Shed, Ephemeris, bioblend, docker-galaxy-stable, Pulsar, and Planemo).

**Prerequisites:**

* Your interest.

