---
title: 'June 2019 Galactic News'
tease: 'GCC2019 registration / Poster/Demo abstracts / sponsors, GTN CoFest, Platforms, Pubs, Jobs, Blog!, Training, Tools, Releases and more!'
date: ""
---

[<img class="float-right" src="/src/news/2019-06-galaxy-update/2019-06-galaxy-update-image.png" alt="Galaxy News" width="350" />](/src/galaxy-updates/index.md>)

The *June 2019 Galactic News* is here! This is a summary of what is going on in the Galaxy community.

* Events
  * **[GCC2019](/src/news/2019-05-galaxy-update/index.md#gcc2019-registration):**
    * **Advance registration ends 7 June**
    * Poster and Demo Abstract Submission closes 10 June
    * *GigaScience* returns as a GCC sponsor!
  * [GTN CoFest and Community Call: 16 May](/src/news/2019-05-galaxy-update/index.md#gtn-cofest-and-community-call)
  * Plus [XXXX other upcoming events](/src/news/2019-05-galaxy-update/index.md#upcoming-events) in the next 90 days
* [150 new publications](/src/news/2019-05-galaxy-update/index.md#publications), great resources lead to great insight. 
* Some most excellent [Galaxy Platform News](/src/news/2019-05-galaxy-update/index.md#galaxy-platforms-news), including a single cell 'omics workbench, network analysis, subcellular spatial organization modeling, and more!
* A new entry to [The Galactic Blog](/src/news/2019-05-galaxy-update/index.md#new-galactic-blog-post), Galaxy hearts GCP.  
* At least [XXXX Open positions](/src/news/2019-05-galaxy-update/index.md#whos-hiring) in six countries on two continents. 
* [Updates to training materials](/src/news/2019-05-galaxy-update/index.md#doc-hub-and-training-updates).
* [New releases](/src/news/2019-05-galaxy-update/index.md#releases).
* And [some other news](/src/news/2019-05-galaxy-update/index.md#other-news) too.

If you have anything to add to next month's newsletter, then please send it to outreach@galaxyproject.org.

----

# GCC2019 Registration 

**[Advance registration](/src/events/gcc2019/registration/index.md) for the [2019 Galaxy Community Conference (GCC2019)](/src/events/gcc2019/index.md) is still open! Discounted registration rates are available until 07JUN19.**

[<img class="float-right" style="max-width: 300px" src="/src/events/gcc2019/gcc2019-logo-big.png" alt="GCC2019" />](/src/events/gcc2019/index.md)

GCC2019 will be held 1-6 July in Freiburg, Germany.  The *tenth* [GCC](/src/events/gcc2019/index.md) will have many familiar features from earlier years, including accepted and lightning talks, posters and demos, birds-of-a-feather gatherings (BoFs), training, and a CollaborationFest.  GCC2019 brings the [most significant conference program update](https://gcc2019.sched.com/) in several years:

* some training will be integrated with the main conference,
* three days of conference instead of two,
* several parallel sessions,
* and a strong emphasis on organizing content by domain.

If you are working in data intensive life science research then there will not be a better place to share your work, learn from others, and find new collaborators.

Please find additional information on GCC2019 [here](/src/events/gcc2019/index.md).

## Poster & Demo Abstract Submission Ends 7 June

Abstracts for [poster presentations and software demonstrations](/src/events/gcc2019/abstracts/index.md) are still being accepted for consideration. 
Discounted registration rates are available until 07JUN19.** Got something to show? [Submit your abstract before the 10JUN19 deadlne!](/src/events/gcc2019/abstracts).

## Sponsor: *GigaScience*

<img class="mx-auto" src="" alt="GigaScience Journal" />](http://www.gigasciencejournal.com/)

[<img class="float-right" style="max-width: 200px" src="/src/images/logos/GigaScienceLogo250.png" alt="GigaScience" />](https://academic.oup.com/gigascience)

***[GigaScience](https://academic.oup.com/gigascience)*** is an online open access, open data, open peer-review journal published by [Oxford University Press](https://academic.oup.com/) and [BGI](https://www.bgi.com/us/). The journal offers ‘big data’ research from the life and biomedical sciences, and on top of 'Omics research includes the growing range of work that uses difficult-to-access large-scale data, such as imaging, neuroscience, ecology, systems biology, and other new types of shareable data. GigaScience is unique in the publishing industry as it publishes all research objects (data, software tools, source code, workflows, containers and other elements related to the work underpinning the findings in the article). Promoting Open Science, all published software needs to be under an OSI-license, all supporting data must be available and open, and all peer review is carried out transparently. Presenting workflows via our [GigaGalaxy.net server](http://gigagalaxy.net/), *novel work from GCC2019 is eligible to a 15% APC if it is submitted to our Galaxy series.*

# Upcoming events

These and [other Galaxy related events](/src/events/index.md) are coming up:


----

# Publications

**150 new publications** referencing, using, extending, and implementing Galaxy were added to the [Galaxy Publication Library](https://www.zotero.org/groups/galaxy) in the last month.

## Highlighted Publications

There were 10 *[Galactic](https://www.zotero.org/groups/1732893/galaxy/items/order/dateAdded/sort/desc/tag/%2BGalactic)* and *[Stellar](https://www.zotero.org/groups/1732893/galaxy/items/order/dateAdded/sort/desc/tag/%2BStellar)* publications added, and 9 of those are open access:

* **[Accessible and reproducible mass spectrometry imaging data analysis in Galaxy]( https://doi.org/10.1101/628719)**, Melanie Christine Föll, Lennart Moritz, Thomas Wollmann, Maren Nicole Stillger, Niklas Vockert, Martin Werner, Peter Bronsert, Karl Rohr, Björn Andreas Grüning, Oliver Schilling. *bioRxiv* 628719; doi: 10.1101/628719
* **[G-OnRamp: a Galaxy-based platform for collaborative annotation of eukaryotic genomes](https://doi.org/10.1093/bioinformatics/btz309)**, Yating Liu, Luke Sargent, Wilson Leung, Sarah C. R. Elgin and [Jeremy Goecks](/src/people/jeremy-goecks/index.md). *Bioinformatics*, btz309, doi: 10.1093/bioinformatics/btz309 
* **[MetaDEGalaxy: Galaxy workflow for differential abundance analysis of 16s metagenomic data](https://f1000research.com/articles/8-726)**, Mike W.C. Thang, Xin-Yi Chua, Gareth Price, Dominique Gorse, Matt A. Field. *F1000Research* 2019, 8:726, doi: 10.12688/f1000research.18866.1)
* **[NGPhylogeny.fr: new generation phylogenetic services for non-specialists](https://doi.org/10.1093/nar/gkz303)**, Frédéric Lemoine, Damien Correia, Vincent Lefort, Olivia Doppelt-Azeroual, Fabien Mareuil, Sarah Cohen-Boulakia, Olivier Gascuel. *Nucleic Acids Research*, gkz303, doi: 10.1093/nar/gkz303
* **[Rice Galaxy: an open resource for plant science](https://academic.oup.com/gigascience/article/8/5/giz028/5492234)**, Venice Juanillas, Alexis Dereeper, Nicolas Beaume, Gaetan Droc, Joshua Dizon, John Robert Mendoza, Jon Peter Perdon, Locedie Mansueto, Lindsay Triplett, Jillian Lang, Gabriel Zhou, Kunalan Ratharanjan, Beth Plale, Jason Haga, Jan E. Leach, Manuel Ruiz, Michael Thomson, Nickolai Alexandrov, Pierre Larmande, Tobias Kretzschmar and Ramil P. Mauleon. *GigaScience*, 8, 2019, 1–14, doi: 10.1093/gigascience/giz028
* **[The RNA workbench 2.0: next generation RNA data analysis](https://doi.org/10.1093/nar/gkz353)**, Jorg Fallmann, Pavankumar Videm, Andrea Bagnacani, Bérénice Batut, Maria A. Doyle, Tomas Klingstrom, Florian Eggenhofer, Peter F. Stadler, Rolf Backofen and [Björn Grüning](https://github.com/bgruening). *Nucleic Acids Research*, gkz353, doi: 10.1093/nar/gkz353
* **[Galaxy External Display Applications: Closing a dataflow interoperability loop](http://dx.doi.org/10.1101/642280)**, [Daniel Blankenberg](/src/people/dan/index.md), [John Chilton](/src/people/john-chilton/index.md), and [Nate Coraor](/src/people/nate/index.md). *bioRxiv* 642280; doi: 10.1101/642280
* **[Prediction and Characterization of miRNA/Target Pairs in Non‐Model Plants Using RNA‐seq]()**, Kira C. M. Neller, Alexander Klenov, Katalin A. Hudak. *Current Protocols in Plant Biology*, e20090. doi: 10.1002/cppb.20090
* **[The Repetitive DNA Composition in the Natural Pesticide Producer *Tanacetum cinerariifolium:* Interindividual Variation of Subtelomeric Tandem Repeats](https://doi.org/10.3389/fpls.2019.00613)**, Jelena Mlinarec, Ana Skuhala, Adela Jurković, Nenad Malenica, Jamie McCann, Hanna Weiss-Schneeweiss, Borut Bohanec and Višnja Besendorfer. *Frontiers in Plant Science*, 16 May 2019, doi: 10.3389/fpls.2019.00613

## Publication Topics


----

# Galaxy Platforms News

The [Galaxy Platform Directory](/src/use/index.md) lists resources for easily running your analysis on Galaxy, including publicly available servers, cloud services, and containers and VMs that run Galaxy.  A lot was added in March:

## The TGM pipeline

[<img class="float-right" style="max-width: 110px" src="/src/use/tgm/tgm-fig1.png" alt="tgmpipeline" />](/src/use/tgm/index.md)

The *[TGM pipeline](http://r3lab.uni.lu/web/tgm-pipeline/)* is a platform combining [tranSMART server](http://r3lab.uni.lu/web/tgm-pipeline/#transmart-setup), [Galaxy Server](http://tgm-pipeline.uni.lu/galaxy/), and [MINERVA platform](http://r3lab.uni.lu/web/tgm-pipeline/#minerva-setup) to enable visually-aided exploration, analysis, and interpretation of high-throughput translational medicine data. The TGM pipeline enables establishing workflows with Galaxy for data processing using tools and resources of the tranSMART server. 


## ARGA

[<img class="float-right" style="max-width: 110px" src="/src/use/arga/arga-pipeline.png" alt="arga" />](/src/use/arga/index.md)

The [Antibiotic Resistance Gene Analyzer (ARGA)](http://mem.rcees.ac.cn:8083/) is a pipeline used for identification of antibiotic resistance genes using the Sequence Database for ARGS (SDARGS). The ARGA Galaxy instance provides all of the nucleotide and protein sequences in SDARG and ARGs primer databases in the Shared Data and Data Libraries sections for users. A guide for recommended usage of ARGA can be found [here](http://mem.rcees.ac.cn:8083/static/SDARG_Galaxy_pipeline_user_guide.pdf).  


## NGPhylogeny.fr

[<img class="float-right" style="max-width: 110px" src="/src/use/ngphylogeny/ngphylogeny-steps.png" alt="ngphylogeny" />](/src/use/ngphylogeny/index.md)

[NGPhylogeny.fr](https://ngphylogeny.fr/) is a free and simple to use web service dedicated to reconstructing and analysing phylogenetic relationships between molecular sequences. The NGPhylogeny service allows users to execute and customize Galaxy workflows while employing a novel interface to Galaxy. Additional documentation can be found [here](https://ngphylogeny.fr/documentation).


## NanoGalaxy

[<img class="float-right" style="max-width: 110px" src="/src/use/nanogalaxy/nanogalaxy-toolkit.png" alt="nanogalaxy" />](/src/use/nanogalaxy/index.md)

[NanoGalaxy](https://nanopore.usegalaxy.eu/) is a webserver for processing, analysis, and visualization of long-read nucleic acid sequencing data from Oxford Nanopore Technologies (ONT) and Pacific Biosystems (PacBio) with Galaxy. This specialized Galaxy instane provides access to a collection of best practice workflows and commonly used tools for analysis of long-read sequence data.

## Galaxy Workbench for Mass Spectrometry Imaging

This is a [custom docker image](https://hub.docker.com/r/mcfoell/galaxy-msi) for doing mass spectrometry image processing in a Galaxy platform.  The tools available in this image are also available on the [UseGalaxy.eu](https://usegalaxy.eu/) platform.

## Rice Galaxy

[<img class="float-right" style="max-width: 110px" src="/src/use/rice-galaxy/rice-galaxy-3k.jpg" alt="Rice Galaxy" />](/src/use/rice-galaxy/index.md)

The [Rice Galaxy server](http://galaxy.irri.org/) is a revamped version of IRRI's previously existing Galaxy server.  It features tools for designing single-nucleotide polymorphism assays, analyzing genome-wide association studies, population diversity, rice−bacterial pathogen diagnostics, and a suite of published genomic prediction methods and has shared datasets that include high-density genotypes from the 3,000 Rice Genomes project and sequences with corresponding annotations from 9 published rice genomes.


## Galaxy Platforms in Publications

We tag papers that use, mention, implement or extend public Galaxy platforms (servers, services, clouds, containers...).  Here are the platforms that were referenced at least twice in the past month's publications:



----

[<img class="float-right" src="/src/blog/2019-04-gcp/ggcp.png" alt="ggcp" style="max-width: 120px" />](/src/blog/index.md)

# New Galactic Blog Post

This month we have a *[Galactic Blog](/src/blog/index.md)* post where [Enis Afgan](https://github.com/afgane) and [Vahid Jalili](https://github.com/VJalili) describe recent efforts to [Build up support for the Google Cloud Platform in Galaxy](/src/blog/2019-04-gcp/index.md).

----

# Who's Hiring

<div class='float-right'><img src="/src/images/GalaxyIsExpandingCloud.png" alt="Please Help! Yes you!" width="220" /></div>

The *dark energy* of irreproducible research is threatening the science universe! Please help the Galaxy push it back!

* [Postdoctoral Fellows](https://www.postdocjobs.com/posting/7057197), [Blankenberg Lab](/src/people/dan/index.md), Cleveland Clinic, Cleveland, Ohio, United States.
* [Scientist - Molecular R&D](https://jobs.vibrantm.com/emd/job/Rockville-Scientist-Molecular-R&D-MD-20850-3304/518778201/), MilliporeSigma, Rockville, Maryland, United States
* ELIXIR Belgium has **three Galaxy-related openings** in Ghent:
  * [ELIXIR Open Science Community Manager](https://vibvzw.jobsoid.com/j/17164/elixir-open-science-community-manager), VIB-UGent Center for Plant Systems Biology
  * [ELIXIR Software developer data management tools](https://vibvzw.jobsoid.com/jobs/35bba14f-79b7-4388-b015-a4fb9d97ee41/17168/software-developer-data-management-tools), VIB-UGent Center for Plant Systems Biology
  * [ELIXIR Scientific Cloud Coordinator](https://vibvzw.jobsoid.com/j/17165/elixir-scientific-cloud-coordinator), VIB-UGent Center for Plant Systems Biology
* [Software Developer](https://sjobs.brassring.com/TGnewUI/Search/home/HomeWithPreLoad?partnerid=25240&siteid=5341&PageType=JobDetails&jobid=1433370#jobDetails=1433370_5341), Harvard T.H. Chan School of Public Health, Boston, Massachusetts, United States. "Basic automated analysis workflows using Galaxy for 16S marker gene, metagenomic, and metatranscriptomic data leveraging existing software." 
* The [The European Galaxy Team has open positions](https://usegalaxy-eu.github.io/posts/2019/01/10/openpositions/), Freiburg, Germany
  * Software engineer, system analysts/administrators, data analyst, and a community and/or research manager

Have a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/src/news/index.md) and include it in next month's [update](/src/galaxy-updates/index.md).

---

# Doc, Hub, and Training Updates

[<img class="float-right img-responsive" style="max-width: 15em;" src="/src/images/galaxy-logos/GTNLogoTrans300.png" alt="Galaxy Training Materials" />](https://galaxyproject.github.io/training-material/)

Updates from the [Galaxy Training Materials](https://galaxyproject.github.io/training-material/):

* A [short version of the "16S Microbial Analysis with mothur"](https://galaxyproject.github.io/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html) was added.  This complements the [longer version](https://galaxyproject.github.io/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html). You can switch at any point from the extended one to the short one. Which is great if you are running out of time.
* Tired of individually loading training datasets from Zenodo?  Tutorials can now support [bulk copying of URL lists](https://github.com/galaxyproject/training-material/pull/1383).  Thanks to [Helena Rasche](https://github.com/erasche).

Hub updates:

* [Data Privacy in Galaxy](/src/learn/privacy-features/index.md): A new page describing how to use Galaxy's data privacy features; from [Martin Cech](/src/people/martin/index.md)


---

# ToolShed Contributions

<div class='right'>[<img src="/src/images/galaxy-logos/galaxy-toolshed-300.png" alt="Galaxy ToolShed" width="150"  />](http://toolshed.g2.bx.psu.edu/)</div>

[Tool Shed contributions in April and May 2019](/src/toolshed/contributions/2019-05/index.md).

# Releases

New additions to (and editions in) the Galaxy Ecosystem.

## Planemo 0.60.0

[<img class="float-right" src="/src/images/galaxy-logos/planemo-logo.png" alt="Planemo" width="200" />](https://pypi.python.org/pypi/planemo/)

[Planemo](https://pypi.python.org/pypi/planemo) is a set of command-line utilities to assist in building tools for the Galaxy project.  These releases included [numerous fixes and enhancements](https://github.com/galaxyproject/planemo/blob/master/HISTORY.rst).

See [GitHub for details](https://github.com/galaxyproject/planemo/blob/master/HISTORY.rst).

## Pulsar 0.11.0

[<img class="float-right" src="/src/images/galaxy-logos/pulsar_transparent.png" alt="Pulsar" width="170" />](https://pypi.python.org/pypi/pulsar-app/)

[Pulsar](https://pypi.python.org/pypi/pulsar-app/) is a Python server application that allows a Galaxy server to run jobs on remote systems (including Windows) without requiring a shared mounted file systems. Unlike traditional Galaxy job runners - input files, scripts, and config files may be transferred to the remote system, the job is executed, and the results are transferred back to the Galaxy server - eliminating the need for a shared file system.

## ephemeris 0.10.0

[Ephemeris](https://github.com/galaxyproject/ephemeris) is a small Python library and set of scripts for managing the bootstrapping of Galaxy plugins - tools, index data, and workflows. It has [extensive documentation](https://ephemeris.readthedocs.org/).  This [release](https://github.com/galaxyproject/ephemeris/blob/master/HISTORY.rst) features numerous newly contributed features.


----

# Other News

[<img class="float-right" src="/src/news/2019-06-galaxy-update/galaxy-status.png" alt="Galaxy Status web site" width="220" />](https://status.galaxyproject.org/)

## Galaxy Services Status Web Site

Automated status tracking of our most important services is available at  [status.galaxyproject.org](https://status.galaxyproject.org/).  This is updated in real time, and will be used to post status updates whenever there are issues.


## Black Duck Open Hub updates it's Galaxy project stats

[Black Duck](http://www.blackducksoftware.com/)'s [Open Hub](https://www.openhub.net/) web site reports stats on open source projects such as Galaxy.  They just updated [their summary of Galaxy](https://www.openhub.net/p/galaxybx):

> Over the past twelve months, 198 developers contributed new code to Galaxy Bioinformatics Platform. This is one of the largest open-source teams in the world, and is in the top 2% of all project teams on Black Duck's Open Hub.

[Take a look](https://www.openhub.net/p/galaxybx).

## data.galaxyproject.org directory view.

From [Enis Afgan](/src/people/enis-afgan/index.md):

> If you ever wondered what reference data is available on the Galaxy Project CVMFS at data.galaxyproject.org, here's a snapshot [file system directory view](https://gist.github.com/afgane/b527eb857244f43a680c9654b30deb1f).

