---
title: "Cancer informatics working group T2 update"
tease: "CI-WG Updates from the tertile"
authors: "Luke Sargent"
date: "2021-09-02"
---

# Cancer Informatics + Galaxy
Over this last [tertile](https://en.wiktionary.org/wiki/tertile) (or, if you prefer, [quadrimestre](https://en.wiktionary.org/wiki/quadrimestre)), the Cancer Informatics working group has continued its efforts to bring cancer informatics tools, workflows, and visualizations to the Galaxy community, with the downstream goal of launching an entire 'flavor' of Galaxy (Galaxy Cancer) dedicated specifically to cancer data analysis. 

Some highlights of these efforts include:

## MCMICRO Tools on GVL
The GVL, or [Genomics Virtual Laboratory](https://www.gvl.org.au/), is Galaxy Cancer's chosen platform. It provides [kubernetes](https://kubernetes.io/)-based cloud-agnostic resource provisioning and management, and allows Galaxy to be integrated with several bundled services -- [Grafana](https://grafana.com/) for service stats, [Rancher](https://rancher.com/) for cluster management, and [Keycloak](https://www.keycloak.org/) for identity management, to name a few.

A primary goal of the working group is to perform cancer tissue analysis using Galaxy; our focus currently is on [MCMICRO](https://mcmicro.org/), an end-to-end processing pipeline to transform large, multi-channel whole slide images into single-cell data. Work has been done to [adapt this pipeline for Galaxy](https://mcmicro.org/galaxy/), and to provide whole-tissue- and tumor-microarray-based workflows.

Adapting these tools to GVL has been a major focus of this last tertile. As a) they are under continuous development and as such can change, and b) they are currently [Conda](https://docs.conda.io/en/latest/)-based tools, but being are being transitioned to container-based tools, it is unfortunately a bit more complicated than a simple toolshed install. Additionally, quirks of Galaxy-for-kubernetes architecture has presented some challenges, but we have managed over the course of the tertile's testing to arrive at a container-building strategy that integrates with GVL's quirks and prevents the various errors and failures generated by our initial attempts.

<div style="text-align:center;width:600px;margin-left: auto;margin-right: auto;">

![A completed MCMICRO run showing a map of cell-states](mcmicro.png)

*Figure 1: MCMICRO on GVL*

</div>

## oncogene.iobio
Along with new tools and workflows, new visualizations are being brought into the Galaxy fold by way of [Interactive Tools](https://training.galaxyproject.org/training-material/topics/admin/tutorials/interactive-tools/slides.html)(ITs). ITs are a (relatively) new way to let Galaxy incorporate visualization services that might not be suited to Galaxy Display Applications. An IT tool author creates a wrapper much like with any other Galaxy tool (indeed, they can produce outputs and be a component of a workflow!), defines startup behavior and ingresses and the like, specifies a service's container, and *voila*, an IT is born.

An ongoing goal for the working group has been to collaborate with the [iobio team](https://iobio.io/about.html) to port their web-based services to Galaxy, and had previously created a [bam.iobio](https://bam.iobio.io) IT; over this last tertile, we have also added [oncogene.iobio](http://oncogene.iobio.io).

<div style="text-align:center;width:600px;margin-left: auto;margin-right: auto;">

![The oncogene.iobio IT launched from Galaxy](oncogene.png)

*Figure 1: The oncogene.iobio IT launched from Galaxy*

</div>

## Future Developments - spoiler alert!

We still have some steps to take before Galaxy Cancer is ready for users, but the day draws nearer! In the meantime, here's a preview of what's to come:
- programmatic access to controlled [CRDC](https://datacommons.cancer.gov/) data
- more iobio services
- more tools and workflows across cancer informatics

If you have questions, comments or suggestions, please let us know in our gitter channel: https://gitter.im/galaxyproject/wg-cancer-informatics


<br />
<br />
<div style="width:250px;margin-left: auto;margin-right: auto;">

![ ](ci_logo.png)

</div>